### PR TITLE
Don't expose private module methods

### PR DESCRIPTION
**Pull request by TingPing**
This is the common Python convention for private methods so lets not expose these to build files as they are implementation details and their behavior is undefined.

### CODE REVIEW

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -981,6 +981,8 @@ def method_call(self, method_name, args, kwargs):
             fn = getattr(self.held_object, method_name)
         except AttributeError:
             raise InvalidArguments('Module %s does not have method %s.' % (self.modname, method_name))
+        if method_name.startswith('_'):
+            raise InvalidArguments('Function %s in %s module is private.' %(method_name, self.modname))
```
Comment (nirbheek) [inline mesonbuild/interpreter.py]: Please use `{!r}` + `.format()` here:

```
InvalidArguments('Function {!r} in module {!r} is private.'.format(method_name, self.modname))
```

I think we should use `.format()` everywhere consistently and change the surrounding code too when we touch it.

File: mesonbuild/modules/gnome.py [hunk 2]
```diff
@@ -232,14 +235,14 @@ def get_dependencies_flags(self, deps, state, depends=None):
             if hasattr(dep, 'held_object'):
                 dep = dep.held_object
             if isinstance(dep, dependencies.InternalDependency):
-                cflags.update(self.get_include_args( state, dep.include_directories))
+                cflags.update(self._get_include_args( state, dep.include_directories))
```
Comment (nirbheek) [inline mesonbuild/modules/gnome.py]: Please remove the typo space before `state` while you're at it :)

**Comment by jpakkane** [comment]
Looks good but merge conflicts now...

**Comment by nirbheek** [comment]
More conflicts!

**Pull request merged**